File: chirp_control/lib/utils/alert_prefs.dart
import 'package:shared_preferences/shared_preferences.dart';

const _sonarAlertsKey = 'settings_sonar_alerts';
const _dredgeWarningsKey = 'settings_dredge_warnings';

Future<bool> loadSonarAlertsEnabled() async {
  final prefs = await SharedPreferences.getInstance();
  return prefs.getBool(_sonarAlertsKey) ?? true;
}

Future<void> saveSonarAlertsEnabled(bool value) async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.setBool(_sonarAlertsKey, value);
}

Future<bool> loadDredgeWarningsEnabled() async {
  final prefs = await SharedPreferences.getInstance();
  return prefs.getBool(_dredgeWarningsKey) ?? true;
}

Future<void> saveDredgeWarningsEnabled(bool value) async {
  final prefs = await SharedPreferences.getInstance();
  await prefs.setBool(_dredgeWarningsKey, value);
}

File: chirp-control-web/src/utils/alertPrefs.ts
const SONAR_ALERTS_KEY = "settings_sonar_alerts";
const DREDGE_WARNINGS_KEY = "settings_dredge_warnings";

function loadFlag(key: string): boolean {
  const raw = localStorage.getItem(key);
  return raw === null ? true : raw === "true";
}

function saveFlag(key: string, value: boolean): void {
  localStorage.setItem(key, String(value));
}

export function loadSonarAlertsEnabled(): boolean {
  return loadFlag(SONAR_ALERTS_KEY);
}

export function saveSonarAlertsEnabled(value: boolean): void {
  saveFlag(SONAR_ALERTS_KEY, value);
}

export function loadDredgeWarningsEnabled(): boolean {
  return loadFlag(DREDGE_WARNINGS_KEY);
}

export function saveDredgeWarningsEnabled(value: boolean): void {
  saveFlag(DREDGE_WARNINGS_KEY, value);
}
